packages/request-logic/src/actions/cancel.ts

```diff
@@ -60,18 +61,21 @@ function applyActionToRequest(
   requestCopied = Request.pushExtensionsData(requestCopied, action.data.parameters.extensionsData);
   requestCopied.events.push(generateEvent(action, timestamp, signer));
 
+  if (request.state === RequestLogicTypes.STATE.CANCELED) {
+    throw new Error('Cannot cancel an already canceled request');
+  }
+
   if (signerRole === RequestLogicTypes.ROLE.PAYER) {
-    if (request.state !== RequestLogicTypes.STATE.CREATED) {
-      throw new Error('A payer cancel need to be done on a request with the state created');
+    if (Semver.lt(request.version, '2.1.0')) {
+      if (request.state !== RequestLogicTypes.STATE.CREATED) {
+        throw new Error('A payer cancel need to be done on a request with the state created');
+      }
     }
     requestCopied.state = RequestLogicTypes.STATE.CANCELED;
     return requestCopied;
   }
 
   if (signerRole === RequestLogicTypes.ROLE.PAYEE) {
-    if (request.state === RequestLogicTypes.STATE.CANCELED) {
```

Comment:
does this mean we were not throwing if a PAYER was cancelling twice? weird...

Reply:
Yes looks like it. But this can't happen from Invoicing because the cancel/reject action doesn't show if it's already canceled/rejected once.